datafusion/sqllogictest/test_files/array_query.slt

```diff
@@ -0,0 +1,160 @@
+  SELECT * FROM (VALUES
+      ([1,2,3], [4,5], 1)
+   )
+    UNION ALL
+  SELECT * FROM (VALUES
+     ([2,3],   [2,3], 1),
+     ([1,2,3], NULL,  1)
+  )
+;
+
+query ??I rowsort
+SELECT * FROM data;
+----
+[1, 2, 3] NULL 1
+[1, 2, 3] [4, 5] 1
+[2, 3] [2, 3] 1
+
+###########
+# Filtering
+###########
+
+query error DataFusion error: Arrow error: Invalid argument error: Invalid comparison operation: List\(Field \{ name: "item", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: \{\} \}\) == List\(Field \{ name: "item", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: \{\} \}\)
+SELECT * FROM data WHERE column1 = [1,2,3];
+
+query error DataFusion error: Arrow error: Invalid argument error: Invalid comparison operation: List\(Field \{ name: "item", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: \{\} \}\) == List\(Field \{ name: "item", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: \{\} \}\)
+SELECT * FROM data WHERE column1 = column2
+
+query error DataFusion error: Arrow error: Invalid argument error: Invalid comparison operation: List\(Field \{ name: "item", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: \{\} \}\) != List\(Field \{ name: "item", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: \{\} \}\)
+SELECT * FROM data WHERE column1 != [1,2,3];
+
+query error DataFusion error: Arrow error: Invalid argument error: Invalid comparison operation: List\(Field \{ name: "item", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: \{\} \}\) != List\(Field \{ name: "item", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: \{\} \}\)
+SELECT * FROM data WHERE column1 != column2
+
+###########
+# Aggregates
+###########
+
+query error Internal error: Min/Max accumulator not implemented for type List
+SELECT min(column1) FROM data;
+
+query error Internal error: Min/Max accumulator not implemented for type List
+SELECT max(column1) FROM data;
+
+query I
+SELECT count(column1) FROM data;
+----
+3
+
+# note single count distincts are rewritten to use a group by
+query I
+SELECT count(distinct column1) FROM data;
+----
+2
+
+query I
+SELECT count(distinct column2) FROM data;
+----
+2
+
+
+# note multiple count distincts are not rewritten
+query II
+SELECT count(distinct column1), count(distinct column2) FROM data;
+----
+2 2
+
+
+###########
+# GROUP BY
+###########
+
+
+query I
+SELECT count(column1) FROM data GROUP BY column3;
+----
+3
+
+# note single count distincts are rewritten to use a group by
+query I
+SELECT count(distinct column1) FROM data GROUP BY column3;
+----
+2
+
+query I
+SELECT count(distinct column2) FROM data GROUP BY column3;
+----
+2
+
+# note multiple count distincts are not rewritten
+query II
+SELECT count(distinct column1), count(distinct column2) FROM data GROUP BY column3;
+----
+1 1
```

Comment:
This answer is actually incorrect (same root cause as https://github.com/apache/arrow-datafusion/issues/9586) and the answer  will be correct (2,2) after https://github.com/apache/arrow-datafusion/pull/9679 is merged

Reply:
fixed